### PR TITLE
Support `extras` in api-client-example v2

### PR DESCRIPTION
**Pull request by robbydrmwn**
Add logic to support extras:
- Sort possible extras files to last
- If has same filename, set as extras, else document

### CODE REVIEW

File: create-project-async/src/project.py [hunk 1]
```diff
@@ -79,3 +93,18 @@ def __add_headers(self, key, value):
...
+        # Sort file paths ending with .json or .txt to be at the end
+        filepaths.sort(key=lambda x: (x.endswith('.json') or x.endswith('.txt'), x))
+        return filepaths
+
+    def __map_documents(self, filepaths):
+        mapped_documents = {}
+        for filepath in filepaths: 
+            filename = filepath.split('/')[-1].split('.')[0]
+            if filename in mapped_documents: 
+                mapped_documents[filename]["extras"] = filepath
```
Comment (ivanm-gdp) [inline create-project-async/src/project.py:107]: extras disini ngga handle array ya? 🤔 
`extra` aja deh kl gitu namanya

File: create-project-async/src/project.py [hunk 2]
```diff
@@ -26,14 +26,28 @@ def create(self, team_id, operations_path, documents_path):
...
+        sorted_filepaths = self.__sort_possible_extra_files_last(filepaths)
+        mapped_documents = self.__map_documents(sorted_filepaths)
+
+        for key in mapped_documents:
+            upload_document_response = self.__upload_file(filepath=mapped_documents[key]["document"])
             documents = {
                 "document": {
-                    "name": filepath.split('/')[-1],
-                    "objectKey": upload_response["objectKey"]
+                    "name": mapped_documents[key]["document"].split('/')[-1],
```
Comment (ivanm-gdp) [inline create-project-async/src/project.py:37]: `.split('/')`

ref: https://github.com/datasaur-ai/api-client-example/pull/25#discussion_r1448711145

mumpung disenggol, mintol diubah jg pakai os.path.split()

File: create-project-async/src/project.py (hunk 1 above)
Comment (robbydrmwn) [reply create-project-async/src/project.py:107]: so far belum sih kalo pake sort logic begini

**Comment by ivanm-gdp** [comment]
titip tambah sedikit explanation di readme / file markdown terpisah ya @robbydrmwn 🙏🏼 
kalau file terpisah, make sure ada link dari readmenya ya

File: create-project-async/src/project.py (hunk 2 above)
Comment (robbydrmwn) [reply create-project-async/src/project.py]: updated

File: create-project-async/src/project.py (hunk 1 above)
Comment (robbydrmwn) [reply create-project-async/src/project.py]: updated

File: create-project-async/readme.md [hunk 3]
```diff
@@ -20,6 +20,8 @@ In this new mutation, we no longer support uploading files directly to the Graph
 Local files are located under `document/` folder.  
 Every file inside the directory will be uploaded to Datasaur as part of the project creation process.
 
+**Note**: If you want to use pairing files, such as inputfile.jpg with inputfile.txt, ensure that they share the same filename. These paired files are commonly used for OCR / Audio projects with transcription.
```
Comment (robbydrmwn) [inline create-project-async/readme.md:23]: okay or not? @ivanm-gdp

Comment (ivanm-gdp) [reply create-project-async/readme.md:23]: oke sepertinya

File: create-project-async/src/project.py [hunk 4]
```diff
@@ -79,3 +93,18 @@ def __add_headers(self, key, value):
             self.headers[key] = value
 
         return self.headers
+
+    def __sort_possible_extra_files_last(self, filepaths):
+        # Sort file paths ending with .json or .txt to be at the end
+        filepaths.sort(key=lambda x: (x.endswith('.json') or x.endswith('.txt'), x))
```
Comment (ivanm-gdp) [inline create-project-async/src/project.py:99]: ini jadi semua file `txt` dan `json` bakal return (True, <filename>) 

file-file lain bakal (False, <filename>)

gitu ya?

Comment (ivanm-gdp) [reply create-project-async/src/project.py:99]: kok bisa key nya tuple 🤔

Comment (ivanm-gdp) [reply create-project-async/src/project.py:99]: aku masih ga paham sih gimana python bisa sorting by comparing tuples, but it seem to work
image

File: create-project-async/src/project.py [hunk 5]
```diff
@@ -26,14 +26,28 @@ def create(self, team_id, operations_path, documents_path):
...
+        sorted_filepaths = self.__sort_possible_extra_files_last(filepaths)
+        mapped_documents = self.__map_documents(sorted_filepaths)
+
+        for key in mapped_documents:
+            upload_document_response = self.__upload_file(filepath=mapped_documents[key]["document"])
             documents = {
                 "document": {
-                    "name": filepath.split('/')[-1],
-                    "objectKey": upload_response["objectKey"]
+                    "name": os.path.basename(mapped_documents[key]["document"]),
```
Comment (ivanm-gdp) [inline create-project-async/src/project.py:37]: nice bisa pakai [basename](https://docs.python.org/3/library/os.path.html#os.path.basename) langsung ya buat ambil filename

**Review by ivanm-gdp — APPROVED** [review]
tested both prelabeled doc & OCR

File: create-project-async/src/project.py (hunk 4 above)
Comment (robbydrmwn) [reply create-project-async/src/project.py:99]: aku juga masih blm paham detail, mostly trial and error bersama chatgpt 😅

**Comment by robbydrmwn** [comment]
merging ya @ivanm-gdp

**Pull request merged**